routes/user.js

```diff
@@ -7,306 +7,236 @@ const { checkTarget } = require('../tools/checkTarget');
-    );
-
+    const taskIndex = managementMetrics.task_metrics.tasks.findIndex(task => task._id.toString() === taskId);
     if (taskIndex === -1) {
       return res.status(404).json({ message: 'Task not found' });
     }
-
-    // Remove the transaction from the array
     managementMetrics.task_metrics.tasks.splice(taskIndex, 1);
-
-    // Save the changes
     await managementMetrics.save();
-
     const taskMetrics = calculateTaskMetrics(managementMetrics);
-
     managementMetrics.task_metrics.performance_this_week = taskMetrics.thisWeekPerformance;
     managementMetrics.task_metrics.performance_this_month = taskMetrics.thisMonthPerformance;
     managementMetrics.task_metrics.performance_this_year = taskMetrics.thisYearPerformance;
     managementMetrics.task_metrics.total_tasks = taskMetrics.totalTasks;
-
-    // Update weekly and monthly values
     taskMetrics.weeklyPerformance.then((resolvedValue) => {
       managementMetrics.task_metrics.weekly = { ...resolvedValue };
     });
-
     taskMetrics.monthlyPerformance.then((resolvedValue) => {
       managementMetrics.task_metrics.monthly = { ...resolvedValue };
     });
-
     await managementMetrics.save();
-
     checkTarget(managementMetrics);
-
     res.status(200).json({ message: 'Task deleted successfully' });
   } catch (error) {
     console.error(error);
     res.status(500).json({ message: 'Internal Server Error' });
   }
 });
 
+// Fetch announcements
+router.get('/fetchAnnouncements', authMiddleware, async (req, res) => {
+  try {
+    // Implementation for fetching announcements
+    const userId = req.userId;
+    const managementMetrics = await ManagementMetrics.findOne({ userId });
+    if (!managementMetrics) {
+      return res.status(404).json({ message: 'User metrics not found for the user' });
+    }
+    const announcements = managementMetrics.announcements;
+    res.status(200).json({ announcements });
+  } catch (error) {
+    console.error(error);
+    res.status(500).json({ message: 'Internal Server Error' });
+  }
+});
+
+// Fetch tasks
+router.get('/fetchTasks', authMiddleware, async (req, res) => {
+  try {
+    // Implementation for fetching tasks
+    const userId = req.userId;
+    const managementMetrics = await ManagementMetrics.findOne({ userId });
+    if (!managementMetrics) {
+      return res.status(404).json({ message: 'User metrics not found for the user' });
+    }
+    const tasks = managementMetrics.task_metrics.tasks;
+    res.status(200).json({ tasks });
+  } catch (error) {
+    console.error(error);
+    res.status(500).json({ message: 'Internal Server Error' });
+  }
+});
+
 function convertDateFormat(inputDate) {
-  const date = new Date(inputDate);
-  const year = date.getFullYear();
-  const month = (date.getMonth() + 1).toString().padStart(2, '0'); // Ensure two digits for month
-  const day = date.getDate().toString().padStart(2, '0'); // Ensure two digits for day
-  return `${year}-${month}-${day}`;
+  // Date conversion function
```

Comment:
The date conversion function is empty. Thats why the task metrics are not getting calculated.